include/VertexBookkeeping.h

```diff
@@ -26,6 +26,11 @@ struct VertexBookkeeping {
     // Residual capacity of this vertex
     int residual;
 
+    // map to store the marked vertices
+    std::map<VertexPtr, bool> marked;
+
+    bool star;
```

Comment:
This variable should be documented with the use, and also the place where it is updated.